### ISSUE TITLE
AttributeError: module 'jax.numpy' has no attribute 'DeviceArray'

### ISSUE THREAD

**Issue by FMGS666** [google/jax-cfd#226]
Hello, 

I was trying to install locally `jax-cfd` using `conda`. This is what I have done to set up the environment and if I try to import the package in a script I get the error of the description above. Is there anything particular that I am missing about how to build the library? 

```{bash}
(base) user@host:~/$ conda create -y -n jax-cfd-trial python=3.10 
Channels:
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/user/miniconda3/envs/jax-cfd-trial

  added / updated specs:
    - python=3.10


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main 
  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu 
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h5eee18b_5 
  ca-certificates    pkgs/main/linux-64::ca-certificates-2024.3.11-h06a4308_0 
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.38-h1181459_1 
  libffi             pkgs/main/linux-64::libffi-3.4.4-h6a678d5_0 
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-11.2.0-h1234567_1 
  libgomp            pkgs/main/linux-64::libgomp-11.2.0-h1234567_1 
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-11.2.0-h1234567_1 
  libuuid            pkgs/main/linux-64::libuuid-1.41.5-h5eee18b_0 
  ncurses            pkgs/main/linux-64::ncurses-6.4-h6a678d5_0 
  openssl            pkgs/main/linux-64::openssl-3.0.13-h7f8727e_0 
  pip                pkgs/main/linux-64::pip-23.3.1-py310h06a4308_0 
  python             pkgs/main/linux-64::python-3.10.14-h955ad1f_0 
  readline           pkgs/main/linux-64::readline-8.2-h5eee18b_0 
  setuptools         pkgs/main/linux-64::setuptools-68.2.2-py310h06a4308_0 
  sqlite             pkgs/main/linux-64::sqlite-3.41.2-h5eee18b_0 
  tk                 pkgs/main/linux-64::tk-8.6.12-h1ccaba5_0 
  tzdata             pkgs/main/noarch::tzdata-2024a-h04d1e81_0 
  wheel              pkgs/main/linux-64::wheel-0.41.2-py310h06a4308_0 
  xz                 pkgs/main/linux-64::xz-5.4.6-h5eee18b_0 
  zlib               pkgs/main/linux-64::zlib-1.2.13-h5eee18b_0 



Downloading and Extracting Packages:

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate jax-cfd-trial
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) user@host:~/$ conda activate jax-cfd-trial

(jax-cfd-trial) user@host:~/$ pip install jaxlib jax jax-cfd
Collecting jaxlib
  Using cached jaxlib-0.4.26-cp310-cp310-manylinux2014_x86_64.whl.metadata (1.8 kB)
Collecting jax
  Using cached jax-0.4.26-py3-none-any.whl.metadata (23 kB)
Collecting jax-cfd
  Using cached jax_cfd-0.2.0-py3-none-any.whl.metadata (1.4 kB)
Collecting scipy>=1.9 (from jaxlib)
  Using cached scipy-1.13.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
Collecting numpy>=1.22 (from jaxlib)
  Using cached numpy-1.26.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)
Collecting ml-dtypes>=0.2.0 (from jaxlib)
  Using cached ml_dtypes-0.4.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (20 kB)
Collecting opt-einsum (from jax)
  Using cached opt_einsum-3.3.0-py3-none-any.whl.metadata (6.5 kB)
Collecting tree-math (from jax-cfd)
  Using cached tree_math-0.2.1-py3-none-any.whl.metadata (477 bytes)
Using cached jaxlib-0.4.26-cp310-cp310-manylinux2014_x86_64.whl (78.8 MB)
Using cached jax-0.4.26-py3-none-any.whl (1.9 MB)
Using cached jax_cfd-0.2.0-py3-none-any.whl (197 kB)
Using cached ml_dtypes-0.4.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.2 MB)
Using cached numpy-1.26.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
Using cached scipy-1.13.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (38.6 MB)
Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Using cached tree_math-0.2.1-py3-none-any.whl (21 kB)
Installing collected packages: numpy, scipy, opt-einsum, ml-dtypes, jaxlib, jax, tree-math, jax-cfd
Successfully installed jax-0.4.26 jax-cfd-0.2.0 jaxlib-0.4.26 ml-dtypes-0.4.0 numpy-1.26.4 opt-einsum-3.3.0 scipy-1.13.0 tree-math-0.2.1

(jax-cfd-trial) user@host:~/$ conda env export
name: jax-cfd-trial
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - bzip2=1.0.8=h5eee18b_5
  - ca-certificates=2024.3.11=h06a4308_0
  - ld_impl_linux-64=2.38=h1181459_1
  - libffi=3.4.4=h6a678d5_0
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libstdcxx-ng=11.2.0=h1234567_1
  - libuuid=1.41.5=h5eee18b_0
  - ncurses=6.4=h6a678d5_0
  - openssl=3.0.13=h7f8727e_0
  - pip=23.3.1=py310h06a4308_0
  - python=3.10.14=h955ad1f_0
  - readline=8.2=h5eee18b_0
  - setuptools=68.2.2=py310h06a4308_0
  - sqlite=3.41.2=h5eee18b_0
  - tk=8.6.12=h1ccaba5_0
  - tzdata=2024a=h04d1e81_0
  - wheel=0.41.2=py310h06a4308_0
  - xz=5.4.6=h5eee18b_0
  - zlib=1.2.13=h5eee18b_0
  - pip:
      - jax==0.4.26
      - jax-cfd==0.2.0
      - jaxlib==0.4.26
      - ml-dtypes==0.4.0
      - numpy==1.26.4
      - opt-einsum==3.3.0
      - scipy==1.13.0
      - tree-math==0.2.1
prefix: /home/user/miniconda3/envs/jax-cfd-trial


(jax-cfd-trial) user@host:~/$ python -c "import jax_cfd"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax_cfd/__init__.py", line 19, in <module>
    import jax_cfd.base
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax_cfd/base/__init__.py", line 17, in <module>
    import jax_cfd.base.advection
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax_cfd/base/advection.py", line 20, in <module>
    from jax_cfd.base import boundaries
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax_cfd/base/boundaries.py", line 20, in <module>
    from jax_cfd.base import grids
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax_cfd/base/grids.py", line 25, in <module>
    from jax_cfd.base import array_utils
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax_cfd/base/array_utils.py", line 30, in <module>
    Array = Union[np.ndarray, jnp.DeviceArray]
  File "/home/user/miniconda3/envs/jax-cfd-trial/lib/python3.10/site-packages/jax/_src/deprecations.py", line 54, in getattr
    raise AttributeError(f"module {module!r} has no attribute {name!r}")
AttributeError: module 'jax.numpy' has no attribute 'DeviceArray'
``` 

Seems like the package is using outdated dependencies from `jax`. Should I just downgrade the package to an older version? 

Thanks for the help!

**Comment by scaomath** [comment]
`jax.numpy.DeviceArray` has been deprecated since 4.14, use `jax.Array` instead.

See https://github.com/google/jax/blob/main/CHANGELOG.md

You can either pip install from git, or change your local installed files manually.

**Comment by FMGS666** [comment]
Hello, I proceeded with the github installation eventually, thanks. I wanted to share the problem for possible further reference.

**Comment by shoyer** [comment]
Thanks for the report! This should be fixed now in the latest 0.2.1 release.
